File: CS50/week1/mario-more/mario.c
// declaration of header files and function
#include <cs50.h>
#include <stdio.h>
void print_row(int height);

// main function
int main(void)
{
    int height;

    do
    {
        height = get_int("Height: ");
    }

    while ((height < 1) || (height > 8));

    print_row(height);
}

// function to print rows
void print_row(int height)
{
    int i, j, k;

    // loop function
    for (i = 0; i < height; i++)
    {
        // loop to print spaces
        for (j = (height - 1); j > i; j--)
        {
            printf(" ");
        }

        // loop to print left hashes
        for (k = 0; k <= i; k++)
        {
            printf("#");
        }
        // prints a 2-space gap
        printf("  ");
        // loop to print right hasges
        for (k = 0; k <= i; k++)
        {
            printf("#");
        }

        // moves to next line
        printf("\n");
    }
}

File: CS50/week2/week6/sentimental-mario-more/mario.py
# import from cs50
from cs50 import get_int

# main function


def main():

    # while loop to ask for user input
    while True:
        try:
            height = get_int("Height: ")
            # move on to next function if valid
            if (1 <= height <= 8):
                break
        # reprompt user if input is invalid
        except ValueError:
            pass

    print_row(height)

# function to print rows


def print_row(height):
    i = 0

    # loop function
    while i < height:
        # loop to print spaces
        for j in range(height - i - 1):
            print(" ", end="")

        # loop to print left hashes
        for k in range(i + 1):
            print("#", end="")

        # prints a 2-space gap
        print("  ", end="")
        # loop to print right hasges
        for k in range(i + 1):
            print("#", end="")
        # move to next line
        i += 1
        print()


main()
